service/dap/server.go

```diff
@@ -130,6 +130,18 @@ type Server struct {
 	// sendingMu synchronizes writing to net.Conn
 	// to ensure that messages do not get interleaved
 	sendingMu sync.Mutex
+
+	// running tracks the running state according to the server,
+	// which may not correspond to the actual running state of
+	// the process (e.g. if a running command is temporarily interrupted).
+	running   bool
```

Comment:
The remote attach handler will need to know if things are running on entry. It can use internal debugger isRunning or the new high-level isRunningCmd. If they are running, it will need to issue a halt, so it can accept configuration requests for breakpoints and such from the client. That halt should be detected by this new code and exit the auto-resume loop. So as long as we use the help that halts and sets the internal flag, I think we should be ok.